tools/please_wrapper/main.go

```diff
@@ -0,0 +1,50 @@
+// This main package is a first iteration shim for Please
+// that handles first installation and updates itself.
+package main
+
+import (
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	"gopkg.in/op/go-logging.v1"
+
+	"github.com/thought-machine/please/src/cli"
+	"github.com/thought-machine/please/src/core"
+	"github.com/thought-machine/please/src/fs"
+	"github.com/thought-machine/please/src/update"
+)
+
+func main() {
+	cli.InitLogging(cli.Verbosity(logging.WARNING))
+
+	core.MustFindRepoRoot()
+	if err := os.Chdir(core.RepoRoot); err != nil {
+		log.Fatalf("Error moving to '%s' repo root: %s", core.RepoRoot, err)
+	}
+
+	config, err := core.ReadDefaultConfigFiles(nil)
+	if err != nil {
+		log.Fatalf("Error reading config file: %s", err)
+	}
+
+	plzPath := filepath.Join(config.Please.Location, "please")
+
+	if fs.FileExists(plzPath) {
+		out, err := exec.Command(plzPath, "--version").Output()
+		if err != nil {
+			log.Fatalf("Error getting Please version: %s", err)
+		}
+
+		core.PleaseVersion = strings.TrimSpace(strings.TrimPrefix(string(out), "Please version "))
+	}
```

Comment:
Can't we just check if `filepath.Join(config.Please.Lcation, version,  "please")` exists? I think we talked about not needing to change the active version of Please, and just executing Please in place?

Reply:
The update logic relies on `core.PleaseVersion` to be set. In order to keep the same behaviour this shim will set `core.PleaseVersion` to the current `filepath.Join(config.Please.Location, "please")` binary. This guarantees that it works the same way as before. Unless I'm missing something?